src/main/java/rx/Observable.java

```diff
@@ -5592,8 +5592,8 @@ public final void forEach(final Action1<? super T> onNext, final Action1<Throwab
      *         called by the source Observable
      * @see <a href="http://reactivex.io/documentation/operators/ignoreelements.html">ReactiveX operators documentation: IgnoreElements</a>
      */
-    public final Observable<T> ignoreElements() {
-        return lift(OperatorIgnoreElements.<T> instance());
+    public final <R> Observable<R> ignoreElements() {
+        return lift(OperatorIgnoreElements.<R, T>instance());
```

Comment:
Actually, there is no need to add more type parameters but just do an unchecked cast or raw type cast:

```java
@SuppressWarnings("unchecked")
//...
return (Observable)lift(OperatorIgnoreElements.<T>instance());
```

Note that `instance()` does this re-cast as well since the operator is stateless.